common/common.go

```diff
@@ -105,8 +105,11 @@ func SliceRemove(slice []uint32, h uint32) []uint32 {
 }
 
 func FileExisted(filename string) bool {
-	_, err := os.Stat(filename)
-	return err == nil || os.IsExist(err)
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
```

Comment:
If `filename` is a directory, this function will return false, and the caller will attempt to create the file with the same name and will fail.